src/util/dstring.h

```diff
@@ -220,4 +221,19 @@ struct diagnostics_helpert<dstringt>
   }
 };
 
+dstringt get_dstring_number(std::size_t);
+
+/// equivalent to dstringt(std::to_string(value)), i.e., produces a string
+/// from a number
+template <typename T>
+static inline
+  typename std::enable_if<std::is_integral<T>::value, dstringt>::type
+  to_dstring(T value)
+{
+  if(value >= 0 && value <= DSTRING_NUMBERS_MAX)
```

Comment:
This requires a type cast of `DSTRING_NUMBERS_MAX` to `T` to avoid GCC reporting: "comparison of integer expressions of different signedness".